### ISSUE TITLE
Allow to simulate observations with the gray filter

### ISSUE THREAD

**Issue by lucabaldini** [lucabaldini/ixpeobssim#712]
Admittedly, this is in order to have a controlled setup in which to test the analysis of data acquired with the gray filter more than anything else, see issue #710 and #711.

**Comment by lucabaldini** [comment]
```
[lbaldini@pclbaldini ixpeobssim]$ xpobssim.py --config ixpeobssim/config/toy_point_source.py --duration 1000. --irfname ixpe:obssim_gray:v12
```

exits with the error

```
>>> Reading (in memory) /data/work/ixpe/ixpeobssim/ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d1_obssim_gray_v012.arf...
>>> Input file /data/work/ixpe/ixpeobssim/ixpeobssim/caldb/ixpe/xrt/bcf/vign/ixpe_d1_obssim_gray_vign_v012.fits does not exists
Cannot continue, abort.
```

This is to be expected, as we only have arf and mrf files generated with the gray filter, all the others being the same.

This would suggest that we don't want to add identical response files for all the different flavors (rmf, vignetting, psf, etc...) and treat the gray filter in the form of a different irfname, but rather add another option to xpobssim (for simulation) and to the relevant analysis tools.

**Comment by lucabaldini** [comment]
And it is only now that I realize that we have essentially no capability of opening response files with the gray filter within ixpeobssim---looking at the script in ixpeirfgen that we used to backfill the files, we simply set the name by hand.

```
def create_gray_filter_response_files():
    """Create response files appropriate for the use with the gray filter.
    """
    for irf_type in ('arf', 'mrf'):
        for intent in ('obssim', 'obssim_alpha075', 'obssim_alpha075simple'):
            irf_name = 'ixpe:%s:v%d' % (intent, IRF_VERSION)
            for du_id in DU_IDS:
                trans = gray_filter_transparency(gf_tran_file_path(du_id), ENERGY_GRID)
                logger.info('Backfilling grayfilter %s file for DU %d...', irf_type, du_id)
                file_path = irf_file_path(irf_name, du_id, irf_type, caldb_path=IXPEIRFGEN_CALDB)
                logger.info('Opening %s...', file_path)
                with fits.open(file_path) as hdu_list:
                    file_name = gf_tran_file_name(du_id)
                    hdu_list['PRIMARY'].header['COMMENT'] = 'Gray filter transmission from "%s"' % file_name
                    hdu_list['SPECRESP'].header['CBD20001'] = ('FILTER(GRAY)', 'Parameter Boundary')
                    hdu_list['SPECRESP'].data['SPECRESP'] *= trans
                    dest = file_path.replace('_obssim_', '_obssim_gray_')
                    logger.info('Writing response file to %s...', dest)
                    hdu_list.writeto(dest, overwrite=True)
```

**Comment by lucabaldini** [comment]
These are the files that we currently have in the caldb:

```
[lbaldini@pclbaldini ixpeobssim]$ ls ixpeobssim/caldb/ixpe/gpd/*/*/*gray*
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d1_obssim_gray_alpha075simple_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d1_obssim_gray_alpha075_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d1_obssim_gray_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d2_obssim_gray_alpha075simple_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d2_obssim_gray_alpha075_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d2_obssim_gray_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d3_obssim_gray_alpha075simple_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d3_obssim_gray_alpha075_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/arf/ixpe_d3_obssim_gray_v012.arf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d1_obssim_gray_alpha075simple_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d1_obssim_gray_alpha075_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d1_obssim_gray_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d2_obssim_gray_alpha075simple_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d2_obssim_gray_alpha075_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d2_obssim_gray_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d3_obssim_gray_alpha075simple_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d3_obssim_gray_alpha075_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/mrf/ixpe_d3_obssim_gray_v012.mrf
ixpeobssim/caldb/ixpe/gpd/cpf/tow/ixpe_d1_obssim_gray_tow_v012.fits
ixpeobssim/caldb/ixpe/gpd/cpf/tow/ixpe_d2_obssim_gray_tow_v012.fits
ixpeobssim/caldb/ixpe/gpd/cpf/tow/ixpe_d3_obssim_gray_tow_v012.fits
[lbaldini@pclbaldini ixpeobssim]$ 
```

and, it goes without saying, the semantic is less than optimal, as "ixpe:obssim:v12" maps into "ixpe_d*_obssim_gray_v012", while "ixpe:obssim_alpha075:v12" maps into "ixpe_d*_obssim_gray_alpha075_v012", that is, the "gray" in the second case is stuck between the two parts of the intent.

**Comment by lucabaldini** [comment]
Ok, I think do have the hooks in place to load the response files with the gray filter, see commit 
https://github.com/lucabaldini/ixpeobssim/commit/dd5bf243160066563de7df37e1ca349c3e8f4cfb

**Comment by lucabaldini** [comment]
I believe I have a first version of the thing working. One totally unexpected thing that happened is that the first time I did run a simulation the thing crashed complaining about negative values in the count spectrum. At a closer look, this is due to the fact that the effective area with the gray filter rolls down very violently at low energy, which does not work very well with the spline interpolation below 1.05 keV. Setting --emin to 1.05 gets us up and running. 

I am tracking this on issue #713

**Comment by lucabaldini** [comment]
For completeness, 10000 s of the standard toy point source yields some 920 k events per DU, while the same duration with the gray filter (and settting emin to 1.05 keV) corresponds to 150 k events. Doesn't seem unreasonable.

Now moving forward to testing the analysis part.

**Comment by lucabaldini** [comment]
And, at a second look, I overlooked the fact that the `irf_file_path()` method in ixpeobssim.irf.caldb is used to the set the ancillary files in the PHA1* files, so it definitely needs to be aware of the gray filter.

That means that the convoluted logic for handling the gray filter in `_load_irf_base()` need to be moved there.

**Comment by lucabaldini** [comment]
Done and ready for a pull request.

**Comment by lucabaldini** [comment]
Pull request #716

**Comment by lucabaldini** [comment]
Ok, I have run a simple set of simulations, in both the unweighted and weighted flavors, and it looks like everything is in order. (This is for the standard toy point source.)

More specifically, for the spectropolarimetric fit in XSPEC I get

```
Unweighted:
>>> PhoIndex = 2.0066e+00 +/- 2.4762e-03, target = 2.0000e+00 (2.68 sigma)
>>> norm = 9.9514e+00 +/- 3.2743e-02, target = 1.0000e+01 (1.48 sigma)
>>> A = 1.0264e-01 +/- 2.5400e-03, target = 1.0000e-01 (1.04 sigma)
>>> psi = 2.9948e+01 +/- 7.0848e-01, target = 3.0000e+01 (0.07 sigma)

Weighted:
>>> PhoIndex = 2.0093e+00 +/- 2.6089e-03, target = 2.0000e+00 (3.57 sigma)
>>> norm = 9.9811e+00 +/- 3.4823e-02, target = 1.0000e+01 (0.54 sigma)
>>> A = 9.6586e-02 +/- 2.3104e-03, target = 1.0000e-01 (1.48 sigma)
>>> psi = 3.1289e+01 +/- 6.8504e-01, target = 3.0000e+01 (1.88 sigma)
```

while for the polarization cubes I get

```
Unweighted:
Quantity     2.00--2.83 keV     ...      5.66--8.00 keV    
-------- ---------------------- ... -----------------------
  E_MEAN     2.3256318307584003 ...       6.678337598771613
  COUNTS               548040.0 ...                196071.0
      MU    0.19822841257128668 ...      0.4887611077567326
      W2           304896.59375 ...           39015.9921875
   N_EFF             301251.375 ...           155899.484375
  FRAC_W     0.5496886632362601 ...      0.7951175052659496
  MDP_99    0.03943001051803476 ...    0.022229941299139605
       I               303068.5 ...           77990.8515625
   I_ERR      552.1744384765625 ...        197.524658203125
       Q           25634.515625 ...           3474.94140625
   Q_ERR     3939.0804620506105 ...       571.4630992981455
       U           30842.109375 ...        6548.36083984375
   U_ERR      3938.956537090252 ...       571.2901831585781
      QN    0.08458323776721954 ...     0.04455575719475746
  QN_ERR   0.012997327211671983 ...    0.007327309393976659
      UN    0.10176613181829453 ...     0.08396319299936295
  UN_ERR   0.012996918310844749 ...    0.007325092260350047
 QUN_COV -2.857317744515935e-08 ... -2.3996511590409707e-08
      PD     0.1323278844356537 ...      0.0950527936220169
  PD_ERR   0.012996026533166134 ...    0.007324269800719189
      PA     25.134114305953734 ...       31.02348760647954
  PA_ERR      2.814015981075087 ...      2.2086482363966304
 P_VALUE 3.0941156536000584e-23 ...  2.7574758535941425e-37
  CONFID                    1.0 ...                     1.0
  SIGNIF     10.181350679849002 ...      12.975464042971884

Weighted:
Quantity     2.00--2.83 keV     ...      5.66--8.00 keV    
-------- ---------------------- ... -----------------------
  E_MEAN     2.3273098765657334 ...        6.67931138098253
  COUNTS               459679.0 ...                182567.0
      MU    0.23664201876605206 ...      0.5512017447467845
      W2           354725.53125 ...           43629.2734375
   N_EFF           252641.09375 ...            146230.28125
  FRAC_W     0.5496032965395418 ...      0.8009677611507008
  MDP_99   0.036067284471320876 ...     0.02035298268817259
       I            299363.0625 ...             79874.40625
   I_ERR        595.58837890625 ...        208.876220703125
       Q        14057.986328125 ...         3077.6806640625
   Q_ERR     3559.2290618791185 ...       535.8514599430514
       U         36149.16015625 ...            6872.4609375
   U_ERR     3558.6122793721115 ...        535.610468381959
      QN    0.04695965722203255 ...     0.03853150084614754
  QN_ERR   0.011889339426700708 ...    0.006708675345465261
      UN    0.12075357884168625 ...     0.08604083955287933
  UN_ERR   0.011887279110702281 ...    0.006705658214291377
 QUN_COV -2.244506796528268e-08 ... -2.2671656304822973e-08
      PD    0.12956324219703674 ...      0.0942746177315712
  PD_ERR   0.011886932636153983 ...    0.006704943468439061
      PA      34.37476090388132 ...      32.937914498753244
  PA_ERR      2.628952420262967 ...       2.038855123307682
 P_VALUE 1.6037433520070633e-26 ...  1.2198895601428708e-43
  CONFID                    1.0 ...                     1.0
  SIGNIF     10.899070936435114 ...       14.05790841558101
```

**Comment by lucabaldini** [comment]
Interestingly enough, the polarization degree at low energy from the polarization cubes seems to be a little bit too high in both cases, which might explain issue #714 (but I need to look carefully into that, before declaring victory.)

I think we are good for merging the PR.

**Comment by lucabaldini** [comment]
Shipped in 30.6.0
